### ISSUE TITLE
Client NFS cache not up to date

### ISSUE THREAD

**Issue by ocaisa** [ComputeCanada/puppet-magic_castle#87]
This seems to be new behaviour but recently I am seeing that the NFS clients are not reporting up-to-date contents for the exports which leads to strange behaviour like
```
[user01@node1 GROMACS]$ ls mpi_test_3.log
mpi_test_3.log
[user01@node1 GROMACS]$ ls
ion_channel.tpr  job.slurm
```
Yesterday, I started the cluster without updating CentOS7 and did not run into this issue (at least in the time I was testing).

**Comment by ocaisa** [comment]
Seems to be reproducible, if I comment out the `yum upgrade -y` in `puppetagent.yaml`, this triggers a rebuild of the login and compute nodes, afterwards the problem is gone. If I toggle it back again and rebuild them including the upgrade the problem reappears.

**Comment by ocaisa** [comment]
Restricting the update also seems to do the trick:
```
yum -y -x 'kernel*' -x 'kmod*' upgrade
```

**Comment by cmd-ntrf** [comment]
I am able to reproduce the problem, although it still unclear what causes it.

It appears to only affect /home, as if I create file in /scratch or /project, they appear correctly.

**Comment by cmd-ntrf** [comment]
It appears the issue is related to the export of security labels and kernel `3.10.0-1160.11.1.el7`. 

If I downgrade the kernel to 3.10.0-1160.6.1.el7, the files appear.

If I remove `security_label` from the NFS server export, the files in home appear.

**Comment by cmd-ntrf** [comment]
I have filed an issue in CentOS tracker:
https://bugs.centos.org/view.php?id=18013

**Comment by cmd-ntrf** [comment]
Temporary fix in #88.

Will close this for now until CentOS fix the issue.
